File: Assignment_49/Assignment_49.3.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;
typedef struct node **PPNODE;

void InsertFirst(PPNODE first, int no)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    if ((*first) == NULL)
    {
        (*first) = newn;
    }
    else
    {
        newn->next = (*first);
        (*first) = newn;
    }
}

void InsertLast(PPNODE first, int no)
{
    PNODE newn = NULL;
    PNODE temp = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    if ((*first) == NULL)
    {
        (*first) = newn;
    }
    else
    {
        temp = (*first);

        while (temp->next != NULL)
        {
            temp = temp->next;
        }

        temp->next = newn;
    }
}

void Display(PNODE first)
{
    while (first != NULL)
    {
        printf("| %d |->", first->data);
        first = first->next;
    }
    printf("NULL\n");
}


void DisplayDivByThree(PNODE first)
{   

    while (first != NULL)
    {
       if((first->data) % 3 == 0)
       {
            printf("| %d |->", first->data);
       }
       first = first->next;
       
    }
   
}

int main()
{
    PNODE head = NULL;

    InsertFirst(&head, 51);
    InsertFirst(&head, 42);
    InsertFirst(&head, 22);
    InsertFirst(&head, 21);
    InsertFirst(&head, 8);

    Display(head);

    InsertLast(&head, 112);
    InsertLast(&head, 123);
    InsertLast(&head, 113);

    Display(head);

    DisplayDivByThree(head);

    return 0;
}

File: Assignment_49XX/Assignment_49_3.java


class node
{
    public int data;
    public node next;
    
    public node(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private node first;
    private int iCount;

    public SinglyLL() 
    {
        System.out.println("Object Of SinglyLL gets created");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {
        node newn  = null;

        newn = new node(no);

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
    }
    public void Display()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" |->");
            temp = temp.next;
        }

        System.out.println("null\n");
    }

    
    public void CountDivByThree()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {   
            if((temp.data) % 3 == 0)
            {
               System.out.print("| "+temp.data+" |->");
            }

            temp = temp.next;
             
        }
        System.out.println("null\n");
    }
}

class Assignment_49_3
{
    public static void main(String Arr[])
    {
        SinglyLL obj;

        obj = new SinglyLL();

        obj.InsertFirst(51);
        obj.InsertFirst(31);
        obj.InsertFirst(20);
        obj.InsertFirst(72);
        obj.InsertFirst(10);
        obj.InsertFirst(55);

        obj.Display();

        obj.CountDivByThree();
        
        obj = null;

        System.gc();
    }
}